### ISSUE TITLE
SUPPORT:  particle polishing with movies error

### ISSUE THREAD

**Issue by pconesa** [I2PC/scipion#701]
Dear colleagues,

I have a problem running particle polishing.

Every time I try to run it (with movies) I get an error:

In the run file I have the following notification:

run.stdout:
....................................

slurmstepd: Exceeded step memory limit at some point....
.......................................

Log file:

Protocol failed: Command 'srun  `which relion_particle_polish_mpi`  --i Runs/015602_ProtRelionPolish/extra/relion_it032_data.star --o shiny --angpix 1.108 --movie_frames_running_avg 5 --dont_read_old_files --sym c2 --sigma_nb 100 --perframe_highres 3.000 --autob_lowres 20.000 --bg_radius 155 --mask Runs/015357_ProtRelionPostprocess/extra/postprocess_automask.mrc ' returned non-zero exit status 137

Any recommendations?

In the gui is only MPI available (no threads).

What are the working settings?

Than you!

Sincerely,
Dmitry

**Comment by pconesa** [comment]
This is back.....@cossorzano,  are you able to help?.
Dear colleagues,

I'm still on the stage of particle polishing (from the frames)


What are on average the setting that you use for the 


1) MPI

2) nodes/himem - ?

What is on average the memory per/core (in himem mode)? 

I tried 20480; 40960 = error

I have something around 800 000 frames




Sincerely
Dmitry

**Comment by cossorzano** [comment]
I don't have any experience with polishing, it was @josuegbl who implemented this protocol and maybe he can add more on this. However, what we have found is that in version 1.0.1, there was no gain in doing particle polishing if you had done cross-correlation+Optical Flow when aligning the movies.

**Comment by josuegbl** [comment]
@cossorzano is right, in version 1.0.1 the results of  cross-correlation+Optical Flow are the same as particle polishing, and it is a simpler way to process the particles. Unfortunately, particle polishing demands  a huge of RAM memory and time, and depends of your particle dimensions and the number of particles. The only  way to decrease the amount of memory without change the dimensions of the images is decreasing the number of MPI processes (until 1), but, of course, there exist a minimum... In previous projects I needed a worstation (1node) with 512GB of RAM to successfully execute particle polish with 3 MPI process (I dont remember the box size, but no more than 400 pixels). 
@pconesa, I supouse you can close this issue....

**Comment by josuegbl** [comment]
@pconesa, which is the aim of this issue? Dmitry's problem is related to RAM memory, because particle poslishing demand a huge amount of memory. In my last mail (part of this mail is posted above) I explained  what was happening, but, I think, we can't do anything else. The solution of his problem is or reduce the MPI processes, or reduce the dimensions of the images, or look a machine with more RAM or reprocess his data with  cross-correlation+Optical Flow in v1.0 or a combination  of these, as I wrote in my previous comment.
